### PR TITLE
Disabled link cleanup

### PR DESCRIPTION
**Pull request by patrickhlauke**
per https://www.w3.org/TR/html-aria/#docconformance

> It is NOT RECOMMENDED to use `aria-disabled="true"` on an `a` element with an `href` attribute.
>
>NOTE
>If a link needs to be "disabled", remove the `href` attribute.

This PR removes the unnecessary `href="#"`, `tabindex="-1"`, and `aria-disabled="true"` from disabled links in both docs pages and examples. `aria-disabled="true"` *is* kept for disabled link-based buttons (that have `role="button"`) as there it's appropriate to use (you *want* to convey to assistive technologies that this thing you're claiming is a button is also disabled at the moment)

Further, the PR extends the "Link functionality caveat" to show the "proper" way (removing `href` and adding `.disabled` class only) to disable a link, but then explains what to do if that's not possible (and then keeps an example with all the traditional `href="#" tabindex="-1" aria-disabled="true"`, but explains clearly that it's not ideal). Same sort of explanation is also added to the pointer event utilities page

x-ref https://github.com/twbs/bootstrap/pull/34874#issuecomment-911769998

### CODE REVIEW

**Comment by mdo** [comment]
When these callouts get so large, I think it's time to turn them into regular docs text. Especially when there are code snippets or live examples within them.

**Comment by patrickhlauke** [comment]
> When these callouts get so large, I think it's time to turn them into regular docs text. Especially when there are code snippets or live examples within them.

done, removed the callout markup and made them regular docs text

File: site/content/docs/5.1/components/buttons.md [hunk 1]
```diff
@@ -85,18 +85,22 @@ Disabled buttons using the `<a>` element behave a bit different:
 
 - `<a>`s don't support the `disabled` attribute, so you must add the `.disabled` class to make it visually appear disabled.
 - Some future-friendly styles are included to disable all `pointer-events` on anchor buttons.
-- Disabled buttons should include the `aria-disabled="true"` attribute to indicate the state of the element to assistive technologies.
+- Disabled buttons using `<a>` should include the `aria-disabled="true"` attribute to indicate the state of the element to assistive technologies.
+- Disabled buttons using `<a>` <em>should not</em> include the `href` attribute.
```
Comment (mdo) [inline site/content/docs/5.1/components/buttons.md:89]:
```suggestion
- Disabled buttons using `<a>` *should not* include the `href` attribute.
```

File: site/content/docs/5.1/utilities/interactions.md [hunk 2]
```diff
@@ -26,9 +26,12 @@ Bootstrap provides `.pe-none` and `.pe-auto` classes to prevent or add element i
...
-{{< /callout >}}
+The `.pe-none` class (and the `pointer-events` CSS property it sets) only prevents interactions with a pointer (mouse, stylus, touch). Links and controls with `.pe-none` are, by default, still focusable and actionable for keyboard users. To ensure that they are completely neutralized even for keyboard users, you may need to add further attributes such as `tabindex="-1"` (to prevent them from receiving keyboard focus) and `aria-disabled="true"` (to convey the fact they are effectively disabled to assistive technologies), and possibly use JavaScript to completely prevent them from being actionable.
+
+If possible, the simpler solution is:
+
+* for form controls, add the `disabled` HTML attribute.
```
Comment (mdo) [inline site/content/docs/5.1/utilities/interactions.md:33]:
```suggestion
- For form controls, add the `disabled` HTML attribute.
```

File: site/content/docs/5.1/utilities/interactions.md [hunk 3]
```diff
@@ -26,9 +26,12 @@ Bootstrap provides `.pe-none` and `.pe-auto` classes to prevent or add element i
...
+* for links, remove the `href` attribute, making it a non-interactive anchor or placeholder link.
```
Comment (mdo) [inline site/content/docs/5.1/utilities/interactions.md:34]:
```suggestion
* For links, remove the `href` attribute, making it a non-interactive anchor or placeholder link.
```

**Review by mdo — APPROVED** [review]
Couple little copy changes. Thanks for reformatting, looks great otherwise! ❤️

**Pull request merged**